x/data-proxy/types/tx.go

```diff
@@ -49,6 +49,21 @@ func (m *MsgEditDataProxy) Validate() error {
 	return nil
 }
 
+func (m *MsgTransferAdmin) Validate() error {
+	if m.Sender == "" {
+		return ErrEmptyValue.Wrap("empty sender")
+	}
+
+	if m.NewAdminAddress == "" {
+		return ErrEmptyValue.Wrap("empty admin address")
+	}
+	if _, err := sdk.AccAddressFromBech32(m.NewAdminAddress); err != nil {
+		return ErrInvalidAddress.Wrap(err.Error())
```

Comment:
Pretty minor point, but since I spent the whole day on error handling I'd suggest:
```
import errorsmod "cosmossdk.io/errors"

...

errorsmod.Wrapf(err, "invalid new admin address: %s", m.NewAdminAddress)
```
`ErrInvalidAddress` doesn't add much value here. Same applies for line 45

Reply:
I tried wrapping it directly but I couldn't for the life of me figure out how to match the expected error. I took the easier route and just reused the SDK error for an invalid address, but if you know which error I should check on for the bech32 checksum check I'm all ears. :)